### ISSUE TITLE
Arb.stringPattern shrinks irrespectively of pattern's min/max chars specification

### ISSUE THREAD

**Issue by dimsuz** [kotest/kotest#2284]
**Which version of Kotest are you using**
4.5.0

The following generator:

```kotlin
@JvmInline value class NodeId(val id: String)

fun Arb.Companion.nodeId(): Arb<NodeId> {
  return Arb.stringPattern("[A-Za-z]{5,8}").map { NodeId(it) }
}
```
when used in a failing test produces this output:

```kotlin
Attempting to shrink arg NodeId(id=YlIfVqM)
Shrink #1: NodeId(id=YlIf) fail
Shrink #2: NodeId(id=Yl) fail
Shrink #3: NodeId(id=Y) fail
Shrink #4: NodeId(id=) fail
Shrink result (after 4 shrinks) => NodeId(id=)
```

**Expected result**

Probably it shouldn't go lower than `5` chars, because of `{5,8}` pattern above.

I guess this happens because `Arb.stringPattern()` has this in its implementation:

```kotlin
   private fun sampleStringPattern(rs: RandomSource): Sample<String> = synchronized(this) {
      generex.setSeed(rs.random.nextLong())
      sampleOf(generex.random(), StringShrinker)
   }
```
...and `StringShrinker` doesn't know anything about Generex patterns.

Do you think it would be possible to improve this? If so, what would be the best way?
(asking because I'd like to try to help with a PR, although not confident enough that I'll do something decent, but I'd like to at least poke around)

**Comment by sksamuel** [comment]
I'm not sure shrinking could work on regex. The complexity of taking a regex and then trying to make a smaller regex that fits the same pattern is probably out of scope for Kotest.

Interestingly, I spotted this just this morning and the string pattern arb has already been updated to remove the shrinker. 
https://github.com/kotest/kotest/blob/master/kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/stringsjvm.kt
